ovh/resource_ovh_cloud_kubernetes_cluster.go

```diff
@@ -0,0 +1,351 @@
+package ovh
+
+import (
+	"fmt"
+	"k8s.io/client-go/tools/clientcmd"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/ovh/go-ovh/ovh"
+)
+
+func resourceCloudKubernetesCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudKubernetesClusterCreate,
+		Read:   resourceCloudKubernetesClusterRead,
+		Delete: resourceCloudKubernetesClusterDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				err := resourceCloudKubernetesClusterRead(d, meta)
+				return []*schema.ResourceData{d}, err
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				DefaultFunc: schema.EnvDefaultFunc("OVH_PROJECT_ID", nil),
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"control_plane_is_up_to_date": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"is_up_to_date": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"next_upgrade_versions": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"nodes_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+				ForceNew: true,
+			},
+			"region": {
+				Type:     schema.TypeString,
+				Default:  "GRA5",
```

Comment:
A constant may be appropriated for default region name.
New clients are on `"GRA7"`.
```suggestion
				Default:  "GRA7",
```

Reply:
the arguments have to follow the api definition
as the api says that region is mandatory, there shouldnt be a default value in the provider
but a list of accepted values instead in a check function